src/operator/controllers/external_traffic/network_policy_uploader.go

```diff
@@ -80,27 +77,32 @@ func (r *NetworkPolicyUploaderReconciler) Reconcile(ctx context.Context, req ctr
 		ctx, &podList,
 		&client.MatchingLabelsSelector{Selector: selector},
 		&client.ListOptions{Namespace: netpol.Namespace})
-
 	if err != nil {
 		logrus.WithError(err).Errorf("error when reading podlist")
 		return ctrl.Result{}, nil
 	}
 
+	if len(podList.Items) == 0 {
+		logrus.
+			WithField("policy", req.NamespacedName.String()).
+			Debug("Failed to resolve any pods, skipping reporting")
+		return ctrl.Result{}, nil
+	}
+
 	var inputs []graphqlclient.NetworkPolicyInput
 
 	for _, pod := range podList.Items {
 		serviceId, err := r.serviceIdResolver.ResolvePodToServiceIdentity(ctx, &pod)
+		if err != nil {
```

Comment:
We should keep retrying here - an error here is never "expected" (unlike 0 results), so it's reasonable to retry. We do want to make sure we don't miss data. Any flow that is critical has to return an error and be retried, unless we expect this to fail for some reason. If a reconciler gives up due to an error - it is missing data.

Reply:
agreed, turned this into an error